### PR TITLE
Transceivers capabilities attributes and OTSi configured parameters

### PR DESCRIPTION
**Pull request by sergiobelotti**
Added groupings to define:
a) common attributes capabilities related to operational mode
b) common attributes capabilities in addition to operational modes related parameters.
c) configured OTSi parameters related to operational modes parameters
d) configured OTSi parameters in addition  to operational modes related parameters.

The intent is to provide common structure for both topology model and interface model.

### CODE REVIEW

File: ietf-layer0-types-ext.yang [hunk 1]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
    
    The typedefs and groupings will be added here. */
 
-}
+// identity
+
+    identity modulation {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:56]: Editorial (indentation):
```suggestion
  identity modulation {
```

File: ietf-layer0-types-ext.yang [hunk 2]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+      "QPSK (Quadrature Phase Shift Keying) modulation";
+  }
+
+  identity DP-QPSK {
+    base modulation;
+    description
+      "DP-QPSK (Dual Polarization Quadrature
+       Phase Shift Keying) modulation";
+  }
+  identity QAM8 {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:72]: Editorial: please add an empty line to improve readability

File: ietf-layer0-types-ext.yang [hunk 3]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+    base modulation;
+    description
+    "DC DP-QAM16 (Dual Polarization Quadrature 
+    Amplitude Modulation)";
+  } 
+
+  identity FEC {
+    description
+      "Enumeration that defines the type of
+       Forward Error Correction";
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:107]:
```suggestion
      "Base Identity the type of
       Forward Error Correction";
```

File: ietf-layer0-types-ext.yang [hunk 4]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+  }
+  identity reed-solomon {
+    base FEC;
+      description
+  "Reed-Solomon error correction";
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:112]: Editorial (indentation):

```suggestion
    description
      "Reed-Solomon error correction";
```

File: ietf-layer0-types-ext.yang [hunk 5]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+  }
+  identity hamming-code {
+    base FEC;
+      description
+  "Hamming Code error correction";
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:117]: Editorial (indentation):

```suggestion
    description
      "Hamming Code error correction";
```

File: ietf-layer0-types-ext.yang [hunk 6]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+
+  }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:142]: Editorial (remove empty line):

```suggestion
  }
```

File: ietf-layer0-types-ext.yang [hunk 7]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+
+  }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:149]: Editorial (remove empty line):

```suggestion
  }
```

File: ietf-layer0-types-ext.yang [hunk 8]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:156]: Editorial (remove empty line):

```suggestion

  }
```

File: ietf-layer0-types-ext.yang [hunk 9]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:166]: Editorial (remove empty line):

```suggestion
    
  }
```

File: ietf-layer0-types-ext.yang [hunk 10]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+ 
+/* transmitter tuning range (f_tx-min, f_tx-max) */
+
+    leaf min-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description
+        "his parameter indicates the minimum frequency for
+                   this template.";
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:210]: Editorial:

```suggestion
        "This parameter indicates the minimum frequency for
         this transceiver.";
```

File: ietf-layer0-types-ext.yang [hunk 11]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+// GROUPINGS  
+
+/* This grouping represent the list of capabilities attributes not under operational-mode */
+
+
+  grouping common-tranceiver-capabilities {
+    description "Capability of an optical tranceiver";
+
+ 
+/* transmitter tuning range (f_tx-min, f_tx-max) */
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:203]: G.698.2 application codes also defines the minimum and maximum central frequency to be supported (see clause 7.2.2 of G.698.2)

File: ietf-layer0-types-ext.yang [hunk 12]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+      config false;
+      description
+        "his parameter indicates the minimum frequency for
+                   this template.";
+    }
+    leaf max-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description "This parameter indicates the minimum frequency
+                   for this template";
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:216]: Editorial:

```suggestion
      description
        "This parameter indicates the maximum frequency
         for this transceiver.";
```

File: ietf-layer0-types-ext.yang [hunk 13]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+    }
+
+    transmitter-tunability-grid {
+      type identityref {
+        base l0-types:flexi-ch-spc-type
+      }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:222]: In order to allow the OTSi's central frequency not to stay on the G.694.1 grid, the type of this attribute should be a frequency in THz or GHz

File: ietf-layer0-types-ext.yang [hunk 14]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+    }
+
+/* supported transmitter power range [p_tx-min, p_tx_max] */
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:225]: G.698.2 application codes also defines the minimum and maximum mean channel output power to be supported (see clause 7.2.1 of G.698.2)

File: ietf-layer0-types-ext.yang [hunk 15]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+      config false;
+      description "The minimum output power of this interface";
+    }
+    leaf max-channel-output-power {
+      type dbm-t;
+      config false;
+      description "The maximum output power of this interface";
+    }
+
+ /* supported receiver power range [p_rx-min, p_rx_max] */
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:238]: G.698.2 application codes also defines the minimum and maximum mean input power to be supported (see clause 7.4.1 of G.698.2)

File: ietf-layer0-types-ext.yang [hunk 16]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+    }
+
+ /* supported inverse multiplexing capabilities such as max. OTSiG:OTSi cardinality */
+
+    leaf multiplexing-cap {
+      type uint32;
+    config false;
+      description "supported inverse multiplexing capabilities
+       such as max. OTSiG:OTSi cardinality";
+    }
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:263]: I am not sure this is an attribute of the transceiver or of the transponder.

Editorial:

```suggestion
    leaf multiplexing-cap {
      type uint32;
      config false;
      description
        "supported inverse multiplexing capabilities
         such as max. OTSiG:OTSi cardinality";
    }
```

File: ietf-layer0-types-ext.yang [hunk 17]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+
+/* This grouping represent the list of attributes related to operational-mode */
+
+  grouping common-tranceiver-capabilities-bymode {
+    description "Attributes capabilities related to operational 
+    modes of an optical tranceiver";
+
+
+    leaf linecoding-bitrate {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:275]: Editorial:

```suggestion
    leaf line-coding-bitrate {
```

File: ietf-layer0-types-ext.yang [hunk 18]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+      type identyref {
+        base line-coding;
+      }
+      config false;
+      description "Bit rate/line coding of optical tributary signal"
+      reference 
+        "ITU-T G.698.2 section 7"
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:282]:
```suggestion
        "ITU-T G.698.2 section 7.1.2"
```

File: ietf-layer0-types-ext.yang [hunk 19]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+                        for this lane";
+    }
+    leaf max-polarization-dependent-loss {
+           type decimal64 {
+             fraction-digits 2;
+           }
+           units dB ;
+           description "Polarization dependent loss";
+    }
+    leaf-list available-modulation-types {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:317]: Should be a leaf

File: ietf-layer0-types-ext.yang [hunk 20]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+      config false;
+      description "OSNR margin to FEC threshold";
+    }
+    leaf  q-margin {
+      type int32;
+      units "dB";
+      config false;
+      description "Q-Factor margin to FEC threshold";
+    }
+    leaf-list available-baud-rates {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:337]: Should be a leaf

File: ietf-layer0-types-ext.yang [hunk 21]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+         symbol rate or modulation rate 
+         in symbols per second or
+         pulses per second. 
+         It is the number of distinct symbol
+         changes (signal events) made to the 
+         transmission medium
+         per second in a digitally 
+         modulated signal or a line code";
+    }
+    leaf-list available-FEC-types {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:353]: Should be a leaf

File: ietf-layer0-types-ext.yang [hunk 22]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+
+  grouping common-tranceiver-configured-param {
+    description "Capability of an optical tranceiver";
+
+
+    leaf transmit-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description
+        "his parameter indicates the actual configured transmitter
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:392]: Editorial:

```suggestion
        "This parameter indicates the actual configured transmitter
```

File: ietf-layer0-types-ext.yang [hunk 23]
```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
...
+        config false;
+        description "Current total receive power";
+    }
+
+  } // grouping for configured attributes out of mode
+
+
+/* This grouping represent the list of parameters values linked to operational-mode  */ 
+
+  grouping common-tranceiver-configured-param-bymode {
```
Comment (italobusi) [inline ietf-layer0-types-ext.yang:418]: I do not think we need this grouping: we need only the configured mode in the common-tranceiver-configured-param grouping

File: ietf-layer0-types-ext.yang (hunk 2 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:72]: ok, done

File: ietf-layer0-types-ext.yang (hunk 20 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:337]: ok, done

**Comment by sergiobelotti** [comment]
> General comment:
> 
> More references are needed to where attributes and identities are defined

Yes, this is correct. Let's do it step by step to add the needed reference.

File: ietf-layer0-types-ext.yang (hunk 11 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:202]: **2020-08-13 CCAMP call**

This range is needed for operational modes

@sergiobelotti Move these attributes to a new grouping which is used by both operational mode and explicit mode and not used by the ITU-T application code

File: ietf-layer0-types-ext.yang (hunk 14 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:225]: **2020-08-13 CCAMP call**

@sergiobelotti : same resolution as in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/4/files/840635ac33a7a32cddb5adb57b9862f0d2e3209e#r469993534

File: ietf-layer0-types-ext.yang (hunk 15 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:238]: **2020-08-13 CCAMP call**

@sergiobelotti : same resolution as in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/4/files/840635ac33a7a32cddb5adb57b9862f0d2e3209e#r469993534

File: ietf-layer0-types-ext.yang (hunk 13 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:222]: **2020-03-20 CCAMP call**

Should be:
```
            type decimal64 {
              fraction-digits 3;
            }
            units GHz;

```
It may be worthwhile defining a common type for this parameters, such as in https://tools.ietf.org/html/draft-ietf-ccamp-layer0-types-03:

```
    typedef frequency-ghz {
      type decimal64 {
        fraction-digits 3;
      }
      units GHz;
      description
        "The DWDM frequency in GHz, e.g., 193112.500";
      reference
        "RFC6205: Generalized Labels for
        Lambda-Switch-Capable (LSC) Label Switching Routers";
    }

```

File: ietf-layer0-types-ext.yang (hunk 16 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:263]: **2020-08-13 CCAMP**

It is a property of the transponder

@sergiobelotti : remove it

File: ietf-layer0-types-ext.yang (hunk 23 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:418]: **2020-08-13 CCAMP**

@sergiobelotti : remove this grouping

Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang]: solve with commit https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/4/commits/40cd7b90dc7e0a65e6bb0b60800e9ae83e455b6c

File: ietf-layer0-types-ext.yang [hunk 24]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+  }
+  typedef frequency-ghz {
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:200]: having two values for the frequency may lead to inconsistency  ... why is this needed ?

File: ietf-layer0-types-ext.yang [hunk 25]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+    leaf line-coding-bitrate {
+      type identyref {
+        base line-coding;
+      }
+      config false;
+      description "Bit rate/line coding of optical tributary signal"
+      reference 
+        "ITU-T G.698.2 section 7.1.2"
+    }
+    leaf max-polarization-mode-dispersion {
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:247]: I thought we had an agreement on an triplet CD, PMD Penalty here
https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/35
Is this a complementary view ?

File: ietf-layer0-types-ext.yang (hunk 24 above)
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:200]: this is a second definition for frequency and may lead to inconsistency... Why is this needed ?

File: ietf-layer0-types-ext.yang [hunk 26]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+    }
+    leaf available-modulation-type {
+      type identityref {
+      base modulation;
+      }
+      config false;
+    description
+      "Modulation type the specific tranceiver in the list can support";
+    }
+    leaf  osnr-margin {
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:288]: what is the meaning of this value ?  Is this OSNR at limit FEC  eg 20dB? or is this the industrial/EOL margin (eg 1dB) ?

if this is really a margin, I would need the osnr @ FEC limit B2B as well.

File: ietf-layer0-types-ext.yang [hunk 27]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+      config false;
+      description "OSNR margin to FEC threshold";
+    }
+    leaf  q-margin {
+      type int32;
+      units "dB";
+      config false;
+      description "Q-Factor margin to FEC threshold";
+    }
+    leaf-list available-baud-rates {
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:300]: I think it should be a leaf, not a leaf-list, otherwise I do not understand this structure of the operational mode. I would expect that one mode corresponds to one baudrate+one modulation + one bitrate. Othewise how do I know if modes are compatible ?

File: ietf-layer0-types-ext.yang (hunk 24 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:200]: Sometime certain entity like channel spacing/nominal central frequency granularity are defined in GHz sometime in THz. It is not mandatory to have double definiton it is up to us decide to use always one or other measure unit.

File: ietf-layer0-types-ext.yang (hunk 25 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:247]: Correct. Need to update , adding the list of triplets:

list chromatic-and-polarization-dispersion-penalty {
      description
          "Optional penalty associated with a given accumulated CD and PMD.
          This list of triplet cd, pmd, penalty can be used to sample 
          the function penalty = f(CD, PMD).";
      config false;
      uses cd-pmd-penalty ;
     }

File: ietf-layer0-types-ext.yang (hunk 24 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:200]: same comment as above. This is what was already proposed for layer0types .

File: ietf-layer0-types-ext.yang (hunk 27 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:300]: Correct, as for available-modulation-type, need to be a leaf. Need to update accordingly

File: ietf-layer0-types-ext.yang (hunk 26 above)
Comment (ggrammel) [reply ietf-layer0-types-ext.yang:288]: suggested definition: min OSNR: if received OSNR is lower than MIN-OSNR then an increased level of bit-errors post-FEC need to be expected.

Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:288]: > suggested definition: min OSNR: if received OSNR is lower than MIN-OSNR then an increased level of bit-errors post-FEC need to be expected.

Thanks Gert , I like the definition. Moreover , as suggested by Colin, min OSNR be should be done in function of receive power. Use min OSNR at minimum power.

File: ietf-layer0-types-ext.yang [hunk 28]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+           config false;
+           description "Maximum Differential group delay of this mode
+                        for this lane";
+    }
+    leaf max-polarization-dependent-loss {
+           type decimal64 {
+             fraction-digits 2;
+           }
+           units dB ;
+           description "Polarization dependent loss";
```
Comment (egriseri) [inline ietf-layer0-types-ext.yang:278]: Shouldn'tte description be: "Maximum acceptable accumulate polarization dependent loss" ?

File: ietf-layer0-types-ext.yang [hunk 29]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+      }
+      config false;
+    description
+      "Modulation type the specific tranceiver in the list can support";
+    }
+    leaf  osnr-margin {
+      type int32;
+      units "dB";
+      config false;
+      description "OSNR margin to FEC threshold";
```
Comment (egriseri) [inline ietf-layer0-types-ext.yang:292]: We should specify a reference optical bandwidth for the OSNR.

File: ietf-layer0-types-ext.yang [hunk 30]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+      config false;
+      description "Maximum rx optical power";
+    }      
+
+  } // end grouping of attributes capabilities related to operational mode and explicit mode
+
+/* This grouping represent the list of configured parameters */
+/* values independent of operational mode */
+
+  grouping common-tranceiver-configured-param {
```
Comment (egriseri) [inline ietf-layer0-types-ext.yang:415]: typo: should be common-transceiver-configured-param

File: ietf-layer0-types-ext.yang [hunk 31]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+    description "Capability of an optical tranceiver";
```
Comment (egriseri) [inline ietf-layer0-types-ext.yang:416]: Same typo as above : transceiver

File: ietf-layer0-types-ext.yang [hunk 32]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+      "DP-QAM16 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+
+  identity DC-DP-QAM16 {
+    base modulation;
+    description
+    "DC DP-QAM16 (Dual Polarization Quadrature 
+    Amplitude Modulation)";
+  } 
+
```
Comment (ggalimba56) [inline ietf-layer0-types-ext.yang:109]: What does DC stand for ? need to update the description

File: ietf-layer0-types-ext.yang (hunk 24 above)
Comment (ggalimba56) [reply ietf-layer0-types-ext.yang:200]: Sometimes (also in ITU-T) the Grid or n / m Granularity is specified in GigaHz while the central frequency is specified in THz.

File: ietf-layer0-types-ext.yang [hunk 33]
```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
...
+  } // end grouping of attributes capabilities related to operational mode and explicit mode
+
+/* This grouping represent the list of configured parameters */
+/* values independent of operational mode */
+
+  grouping common-tranceiver-configured-param {
+    description "Capability of an optical tranceiver";
+
+
+    leaf transmit-frequency {
```
Comment (ggalimba56) [inline ietf-layer0-types-ext.yang:419]: I'd use "central frequency" to avoid ambiguity

**Comment by ggrammel** [comment]
Hi Sergio,

Thanks for catching this. Suggestion is:

min OSNR: if received OSNR at minimum Rx-power is lower than MIN-OSNR, an increased level of bit-errors post-FEC need to be expected.

File: ietf-layer0-types-ext.yang (hunk 26 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:288]: Done YANG modification

File: ietf-layer0-types-ext.yang (hunk 28 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:278]: yes. I will update accordingly

File: ietf-layer0-types-ext.yang (hunk 29 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:292]: Any suggestion ?

File: ietf-layer0-types-ext.yang (hunk 30 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:415]: done

File: ietf-layer0-types-ext.yang (hunk 31 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:416]: done

**Comment by sergiobelotti** [comment]
> if received OSNR at minimum Rx-power is lower than MIN-OSNR, an increased level of bit-errors post-FEC need to be expected.

Thanks Gert: updated accordingly

File: ietf-layer0-types-ext.yang (hunk 33 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:419]: @ggalimba56 : 👍 updated accordingly

File: ietf-layer0-types-ext.yang (hunk 32 above)
Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang:109]: @ggalimba56 : the meaning is Dual Carrier. Addressed in the code.

File: ietf-layer0-types-ext.yang (hunk 29 above)
Comment (EstherLerouzic) [reply ietf-layer0-types-ext.yang]: I suggest 12.5GHz :)

Comment (egriseri) [reply ietf-layer0-types-ext.yang]: Fine for me :-)

Comment (sergiobelotti) [reply ietf-layer0-types-ext.yang]: So the final description agreed with Enrico would be :  "min OSNR measured over 12.5 GHz resolution bandwidth:
      if received OSNR at minimum Rx-power is lower than MIN-OSNR,
      an increased level of bit-errors post-FEC needs 
      to be expected.

@EstherLerouzic : is it fine for you? If yes I will proceed with merging.
